src/main/java/net/neoforged/art/internal/ClassProviderImpl.java

```diff
@@ -119,7 +119,7 @@ static class ClassInfo implements IClassInfo {
 
             if (!node.fields.isEmpty())
                 this.fields = Collections.unmodifiableMap(node.fields.stream().map(FieldInfo::new)
-                    .collect(Collectors.toMap(FieldInfo::getName, Function.identity())));
+                    .collect(Collectors.toMap(f -> f.getName() + f.getDescriptor(), Function.identity())));
```

Comment:
You forgot to separate the name and descriptor.

Reply:
Seems like same problem for methods as well, before me